tf2onnx/onnx_opset/generator.py

```diff
@@ -44,6 +44,18 @@ def version_1(cls, ctx, node, **kwargs):
             node.set_attr("shape", shape)
             ctx.set_shape(node.output[0], shape)
 
+    @classmethod
+    def version_9(cls, ctx, node, **kwargs):
+        if node.inputs[0].is_const():
+            cls.version_1(ctx, node, **kwargs)
+        else:
```

Comment:
why not move "else:" into version 1 above? That way all versions are fixed (not just 9)?

Reply:
Because "ConstantOfShape" comes since version 9. :-)